src/fit_dataframe.jl

```diff
@@ -269,16 +268,12 @@ function expand_categoricals!(df::DataFrame,categoricals::Array)
     return expand_categoricals!(df, categoricalidxs)
 end
 
-# convert NaNs to NAs
-# isnan(x::NAtype) = false
-isnan(x::AbstractString) = false
-isnan(x::Union{T, Nothing}) where T = isnan(x.value)
 
 # same functionality as above.
 function NaNs_to_Missing!(df::DataFrame)
     m,n = size(df)
     for j=1:n
-        df[!,j] = [ismissing(df[i,j]) || isnan(df[i,j]) ? missing : value for (i,value) in enumerate(df[:,j])];
+        df[!,j] = [(value isa Number && isnan(value)) ? missing : value for value in df[!,j]];
```

Comment:
this is just a bit simpler we only need the `value`, and adds the `isa Number` check